src/Sentry/Laravel/Features/StorageIntegration.php

```diff
@@ -0,0 +1,35 @@
+<?php
+
+namespace Sentry\Laravel\Features;
+
+use Illuminate\Contracts\Filesystem\Filesystem;
+use Illuminate\Contracts\Foundation\Application;
+use Illuminate\Filesystem\FilesystemManager;
+use Sentry\Laravel\Tracing\Storage\TracingFilesystem;
+
+class StorageIntegration extends Feature
+{
+    public function isApplicable(): bool
+    {
+        return $this->isTracingFeatureEnabled('storage');
+    }
+
+    public function setup(): void
+    {
+        $this->container()->afterResolving(FilesystemManager::class, static function (FilesystemManager $filesystemManager): void {
+            $filesystemManager->extend('sentry', function (Application $application, array $config) use ($filesystemManager): Filesystem {
+                $config['driver'] = $config['original_driver'];
```

Comment:
As a sanity check, maybe we should add a check and throw if the `original_driver` is for some reason `sentry` to prevent creating hard to debug infinite loops?

Reply:
I don't think we need to when we automate the driver swap with https://github.com/getsentry/sentry-laravel/pull/726#discussion_r1252201982.

Reply:
I disagree since stupid things can still happen :)

I've reworked it so users can't do uninteded things and we are also a little more cautious.